js-jasmine/src/gilded_rose.js

```diff
@@ -7,59 +7,70 @@ class Item {
 }
 
 class Shop {
-  constructor(items=[]){
-    this.items = items;
+  constructor(items = []) {
+    this.items= items; //instantiates this.items[i]s array
+  }
+
+  removeQuality() {
+    return this.items[i].quality -= this.items[i].quality
+  }
+  decreaseSellIn() {
+    return this.items[i].sellIn -= 1
+  }
+  increaseQuality() {
+    return this.items[i].quality += 1; 
+  }
+  decreaseQuality() {
+   return this.items[i].quality -= 1
+  }
+  isAgedBrie() {
+     this.name == 'Aged Brie'
+  }
+   isBackstagePass() {
+     this.name == 'Backstage passes to a TAFKAL80ETC concert'
+  }
+   isSalfuras() {
+     this.name == 'Sulfuras, Hand of Ragnaros'
```

Comment:
Identified different functions for each item type. Can be extended by moving the code that edits each item type into their own function.
e.g
```
if(isBackstagePass(item) {
    updateBackstage(item)
}
```

Reply:
Makes sense 👍🏿